pkg/apis/federation/v1alpha1/propagatedversion_types.go

```diff
@@ -48,14 +48,14 @@ type PropagatedVersionSpec struct {
 
 // PropagatedVersionStatus defines the observed state of PropagatedVersion
 type PropagatedVersionStatus struct {
-	TemplateVersion string                   `json:"templateversion,omitempty"`
-	OverrideVersion string                   `json:"overrideversion,omitempty"`
-	ClusterVersions []ClusterResourceVersion `json:"clusterversions,omitempty"`
+	TemplateVersion string           `json:"templateversion,omitempty"`
+	OverrideVersion string           `json:"overrideversion,omitempty"`
+	ClusterVersions []ClusterVersion `json:"clusterversions,omitempty"`
 }
 
-type ClusterResourceVersion struct {
-	ClusterName     string `json:"clustername,omitempty"`
-	ResourceVersion string `json:"resourceversion,omitempty"`
+type ClusterVersion struct {
```

Comment:
To me `ClusterVersion` reads as 'version of the cluster'.  Consider using a more descriptive name (e.g. `ClusterObjVersion`).